runhouse/resources/hardware/cluster.py

```diff
@@ -317,7 +318,7 @@ def install_packages(
         from runhouse.resources.envs.env import Env
 
         self.check_server()
-        env = _get_env_from(env) or Env(name=env or Env.DEFAULT_NAME)
+        env = _get_env_from(env, system=self) or Env(name=env or Env.DEFAULT_NAME)
```

Comment:
This makes me a bit nervous - if the env contains package objects they will be serialized and sent back pointing to the cluster's local filesystem. Also not totally clear what's happening here, we're getting the env object from the system by name so we can then extract the name out of it?